pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -606,6 +617,48 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// Reconcile console link in application menu
+	if isClusterDeployment {
+		consoleLinkCrd := &apiextensionsv1beta1.CustomResourceDefinition{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: consoleLinkCrdName}, consoleLinkCrd)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("No support for console links")
+		} else if err != nil {
+			reqLogger.Error(err, "Failed to get ConsoleLink CRD")
+		} else {
+			consoleLinkConfig, err := osutil.NewConsoleLinkConfig(instance, route, configMap)
+			if err != nil {
+				reqLogger.Error(err, "Failed to create ConsoleLinkConfig instance")
+			} else {
+				consoleLink := &consolev1.ConsoleLink{}
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: consoleLinkConfig.Name}, consoleLink)
+				if err != nil && errors.IsNotFound(err) {
+					// If not found, create a console link
+					consoleLink := osutil.NewConsoleLink(consoleLinkConfig)
+					reqLogger.Info("Creating console link", "ConsoleLink.Name", consoleLink.Name)
+					err = r.client.Create(context.TODO(), consoleLink)
+					if err != nil {
+						reqLogger.Error(err, "Failed to create console link", "ConsoleLink.Name", consoleLink.Name)
+					}
+					// Console link created successfully - return and requeue
+					return reconcile.Result{Requeue: true}, nil
+				} else if err != nil {
+					reqLogger.Error(err, "Failed to get console link")
+				} else {
+					// If found and outdated, update it
+					if osutil.IsConsoleLinkOutdated(consoleLink, consoleLinkConfig) {
```

Comment:
Just use the `Patch` operation instead of diffing manually.

Reply:
Can you point to an example?

Reply:
Here: https://github.com/apache/camel-k/blob/e3c94617a4e4d98c26d476bcc93248a1367844b3/pkg/controller/integrationplatform/integrationplatform_controller.go#L154

Reply:
`r.client.Status().Patch` doesn't seem to be available. Also, isn't it supposed to be used to update the status only?

Reply:
You may have to update the Operator SDK dependency. It's both available for the status and the spec.

Reply:
No luck upgrading to 0.11.0
```
$ go get github.com/operator-framework/operator-sdk@v0.11.0
go: finding k8s.io/cloud-provider v0.0.0-00010101000000-000000000000
go: k8s.io/cloud-provider@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go: error loading module requirements
```
Upgrade to 0.12.0 doesn't throw any errors but then it requires Go 1.13 to compile.

Reply:
You can upgrade Go.

Reply:
Done